### PR TITLE
Handle screen rotation and app minimization in LoginActivity

### PR DESCRIPTION
**Pull request by prajwal27**
**Summary**
The data entered in LoginActivity is not handled during screen rotation and app minimization.

Screen Rotation  (Issue 1)
issue1gif

App Minimization (Issue 2)
issue2gif

**Solution**
Handle the data using onSaveInstanceState in LoginActivity
soln-gif

### CODE REVIEW

**Comment by prajwal27** [comment]
Any review on this?

**Comment by sunbiz** [comment]
I don't think we should be saving the username/password unless the connection has been complete and confirmed to be accurate. It should not be stored for example during app minimization.

Is there an issue linked to this PR? like why is this an important issue or even required to be done?

**Comment by prajwal27** [comment]
It's a good issue in terms of UX. We are not saving the username/password anywhere. If the user rotates the phone by mistake and if the credentials are long, the user has to type again.

File: app/src/main/java/org/dhis2/usescases/login/LoginPresenter.java [hunk 1]
```diff
@@ -133,7 +133,7 @@ public void onServerChanged(CharSequence s, int start, int before, int count) {
     @Override
     public void onUserChanged(CharSequence s, int start, int before, int count) {
         isUserNameSet.set(!view.getBinding().userName.getEditText().getText().toString().isEmpty());
-        view.resetCredentials(false, false, true);
+        view.resetCredentials(false, false, false);
```
Comment (sunbiz) [inline app/src/main/java/org/dhis2/usescases/login/LoginPresenter.java:136]: credentials should be reset for both the server change or user change

File: app/src/main/java/org/dhis2/usescases/login/LoginPresenter.java [hunk 2]
```diff
@@ -117,7 +117,7 @@ public void init(LoginContracts.View view) {
     public void onServerChanged(CharSequence s, int start, int before, int count) {
         testingSet = false;
         isServerUrlSet.set(!view.getBinding().serverUrl.getEditText().getText().toString().isEmpty());
-        view.resetCredentials(false, true, true);
+        view.resetCredentials(false, false, false);
```
Comment (sunbiz) [inline app/src/main/java/org/dhis2/usescases/login/LoginPresenter.java:120]: same as when user changes

File: app/src/main/java/org/dhis2/usescases/login/LoginActivity.java [hunk 3]
```diff
@@ -351,13 +365,22 @@ public void onBackPressed() {
     }
 
     @Override
-    public void onSaveInstanceState(Bundle outState, PersistableBundle outPersistentState) {
-        super.onSaveInstanceState(outState, outPersistentState);
+    protected void onSaveInstanceState(@NonNull Bundle outState) {
```
Comment (sunbiz) [inline app/src/main/java/org/dhis2/usescases/login/LoginActivity.java:368]: So this will mean that the previous PersistableBundle won't work?

**Comment by sunbiz** [comment]
close as wont-fix

**Pull request closed without merging**